patches/llvm-all/linux-next/x86/i386-uaccess.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/arch/x86/include/asm/uaccess.h b/arch/x86/include/asm/uaccess.h
+index d8f283b9a..87ef6c80a 100644
+--- a/arch/x86/include/asm/uaccess.h
++++ b/arch/x86/include/asm/uaccess.h
+@@ -318,7 +318,7 @@ do {									\
+ 	__chk_user_ptr(ptr);						\
+ 	switch (size) {							\
+ 	case 1:								\
+-		__get_user_asm(x, ptr, retval, "b", "=q");		\
++		__get_user_asm(x, ptr, retval, "b", "=r");		\
```

Comment:
while this may build, I'm not yet convinced it's the correct fix.  For 32b x86, `=q` limits the register chosen to one of the four that allow accessing the lower byte.  Under register pressure, "=r" allows a different register to be selected.

See https://github.com/ClangBuiltLinux/linux/issues/194#issuecomment-620835960

Reply:
Yes, ClangBuiltLinux/linux#781 needs a proper fix. Also I didn't include this patch: https://github.com/ClangBuiltLinux/continuous-integration/pull/182#issuecomment-619628550. Should I add it?

Reply:
```diff
diff --git a/arch/x86/include/asm/uaccess.h b/arch/x86/include/asm/uaccess.h
index d8f283b9a569..8c6828eee8a1 100644
--- a/arch/x86/include/asm/uaccess.h
+++ b/arch/x86/include/asm/uaccess.h
@@ -314,11 +314,13 @@ do {                                                                      \
 
 #define __get_user_size(x, ptr, size, retval)                          \
 do {                                                                   \
+       unsigned char x_u8;                                             \
        retval = 0;                                                     \
        __chk_user_ptr(ptr);                                            \
        switch (size) {                                                 \
        case 1:                                                         \
-               __get_user_asm(x, ptr, retval, "b", "=q");              \
+               __get_user_asm(x_u8, ptr, retval, "b", "=q");           \
+               (x) = x_u8;                                             \
                break;                                                  \
        case 2:                                                         \
                __get_user_asm(x, ptr, retval, "w", "=r");              \
```
may be more appropriate.

Reply:
Thanks for the patches!